x/dualstaking/README.md

```diff
@@ -0,0 +1,138 @@
+
+1. Call unbond method of the staking module.
+2. Hook on create delegation and unbond the same amount from the providers delegations uniformaly with priority to the empty provider.
+
+#### Validator slashing
+
+1. Hook on calidator slashing
+2. remove the slashed amount uniformaly from the providers delegations
+
+#### Provider delegation
+
+1. Call delegate method of the dualstaking module.
+2. Hook on create delegation and delegate to the empty provider.
+3. redelegate from the empty provider to the actual provider.
+
+#### Provider unbonding
+
+1. redelegate from the the provider to the empty provider.
+2. Call unbond method of the dualstaking module.
+3. Hook on create delegation and unbond from empty provider.
+
+### Hooks
+
+Dual staking module uses [staking hooks](keeper/hooks.go) to achieve its functionality.
+1. AfterDelegationModified: this hook is called whenever a delegation is changed, whether it is created, or modified (NOT when completly removed). it calculates the difference in providers and validators stake to determine the action of the user (delegation or unbonding) depending on who is higher and than does the same with provider delegation.
+    * If provider delegations > validator delegations: user unbonded, uniform unbond from providers delegations (priority to empty provider)
+    * If provider delegations < validator delegations: user delegation, delegate to the empty provider
+2. BeforeDelegationRemoved: this hook is called when a delegation to a validator is removed (unbonding of all the tokens). uniform unbond from providers delegations
+3. BeforeValidatorSlashed: this hook is called when a validator is being slashed. to make sure the balance between validator and provider delegation is kept it uniform unbond from providers delegations the slashed amount. 
+
+### RedelegateFlag
+
+In case of validator redelegation we want to make sure the dual staking module dont take action.
+This is with [antehandler](ante/ante_handler.go), whenever a redelegation msg is running, RedelegateFlag is marked true and the hooks will ignore the delegation changes.
+Note: The RedelegateFlag is an on memory flag (not on state).
+
+## Parameters
+
+The Dualstaking module does not contain parameters.
+
+## Queries
+
+The Dualstaking module supports the following queries:
+
+| Query             | Arguments         | What it does                                  |
+| ----------        | ---------------   | ----------------------------------------------|
+| `params`          | none              | show the params of the module                 |
+| `delegator-providers` | delegator address              | shows the providers that the delegator address is delegated to         |
+| `provider-delegators` | provider address           | shows  all the providers delegators              |
+| `delegator-rewards`       | delegator address           | shows all the claimable rewards of the delegator                             |
+
+## Transactions
```

Comment:
please see other README for a standard Transactions chapter.
Specifically, I'm missing a note about the mandatory `--from` flag, when are the TX applied (immediately? next epoch?) and that the arguments in the table should be: arg (type) like "chain-id (string)"

Reply:
in cosmos it is known that all tx needs from. it does not work without that. feels redundent

Reply:
all changes are made immidiatly, i dont think i should explicitly note it (in contrast to changes in the next epoch)